### PR TITLE
Simplify eval code

### PR DESCRIPTION
**Pull request by neubig**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Currently the eval code is copy-pasted a lot, which leads to duplicated code and lots of difficulty refactoring if we change anything because you need to refactor 13 times. This PR starts to de-duplicate code.

I have de-duplicated 3 of the 13 eval repos, but if this looks good in general I can move on to doing more. One caveat is that this actually will slightly change the prompts for eval :/ The prompts for eval were not very standardized across the different tasks, so this PR standardizes them a bit. Overall, I think this is probably better, as we don't really want to have to require lots of prompt engineering for each different task you do.

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**

Basically it:
1. Splits off shared functionality into a shared directory.
2. Deletes most of the code in the "main" part of the evaluation repos. I'm not changing the instance processing at this time, although that could also probably be de-duplicated to some extent too.

**Other references**

Fixes https://github.com/OpenDevin/OpenDevin/issues/2759

### CODE REVIEW

**Comment by tobitege** [comment]
That is a tedious task to do, but brings a lot of maintainability and readability (back)!

Since we had the other day a PR about applying encoding everywhere, this could be the right time to do it for evals.
I saw that the parser already does it for writing back replies, but in the other occasions it is (yet) without encoding, so there could potentially be some characters end up differently.

However, I just wanted to share a little tidbit as a result of a chat with Sonnet, showing below scenarios.
The other day I had tried with encoding applied and some test steps and indeed ran into an unhandled expection with agentskills' file operations, even though I didn't find out the exact reason.
So from my point of view, the code should expect it to occur and handle it gracefully, obviously.

```py
# This might work, using system default encoding
with open(filename, 'r') as file:
    content = file.read()

# This might raise a UnicodeDecodeError
with open(filename, 'r', encoding='utf-8') as file:
    content = file.read()

# This will always work, but gives you raw bytes
with open(filename, 'rb') as file:
    content = file.read()
    # Now you can decode manually:
    try:
        text = content.decode('utf-8')
    except UnicodeDecodeError:
        # Handle the error, e.g., by ignoring problematic characters
        text = content.decode('utf-8', errors='ignore')
```
The line-endings replacement as in the parser code should still be applied.
So in consequence it would be "rb", "wb", "ab" for the open commands.

I hope this helps the considerations.

**Review by xingyaoww — APPROVED** [review]
Thanks for this!! This is much needed before we drowned ourselves in the sea of code duplication...

**Comment by neubig** [comment]
OK, I think this is done! I'll do a bit of testing on SWE bench and if it works merge it in.
